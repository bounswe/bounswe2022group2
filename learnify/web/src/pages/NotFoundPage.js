import React from 'react';
import {useNavigate} from 'react-router-dom';
import './style.css'
import notfound from '../images/404NotFound.jpg'
import logo from '../images/logo-dblue.png'

function NotFoundPage() {

  const navigate = useNavigate();

  return(        
        <div className='notFoundPageLayout'>
            <div className='notfound-page-logo'>
                <img  style={{float:"left"}} src={logo} alt="Learnify Logo" height={60} />
            </div>
            <div className='notfoundpage-text'><h1>Oops! Something went wrong!</h1></div>
            <img src={notfound} style={{ width: "60%"}} alt="Learnify 404 Not Found" className = 'mockup'/> 
            <div>
                <button className="btn-blue" onClick={() => navigate(-1)}>Return</button>
            </div>
        </div>
    )       
}

export default NotFoundPage;
