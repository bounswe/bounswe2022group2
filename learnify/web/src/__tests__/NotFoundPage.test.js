import React from "react";
import NotFoundPage from "./../pages/NotFoundPage";
import { screen, render, fireEvent } from "@testing-library/react";
import renderer from 'react-test-renderer';
import { BrowserRouter } from "react-router-dom";

describe("NotFoundPage", () => {
    test('should render page layout correctly', async () => {
        const component = renderer.create(<BrowserRouter><NotFoundPage /></BrowserRouter>).toJSON();
        expect(component).toMatchSnapshot();
    });
});

test('Not found page text renders correctly', async () => {
    render(<BrowserRouter><NotFoundPage /></BrowserRouter>);
    const heading = screen.getByText(/Oops! Something went wrong!/i);
    expect(heading).toBeInTheDocument();
}   );

test('not found page illustration renders correctly', async () => {
    render(<BrowserRouter><NotFoundPage /></BrowserRouter>);
    const illustration = screen.getByAltText(/Learnify 404 Not Found/i);
    expect(illustration).toBeInTheDocument();
}   );

test('not found page logo renders correctly', async () => {
    render(<BrowserRouter><NotFoundPage /></BrowserRouter>);
    const illustration = screen.getByAltText(/Learnify Logo/i);
    expect(illustration).toBeInTheDocument();
}   );