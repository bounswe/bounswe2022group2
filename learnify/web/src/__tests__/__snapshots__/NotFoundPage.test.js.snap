// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundPage should render page layout correctly 1`] = `
<div
  className="notFoundPageLayout"
>
  <div
    className="notfound-page-logo"
  >
    <img
      alt="Learnify Logo"
      height={60}
      src="logo-dblue.png"
      style={
        Object {
          "float": "left",
        }
      }
    />
  </div>
  <div
    className="notfoundpage-text"
  >
    <h1>
      Oops! Something went wrong!
    </h1>
  </div>
  <img
    alt="Learnify 404 Not Found"
    className="mockup"
    src="404NotFound.jpg"
    style={
      Object {
        "width": "60%",
      }
    }
  />
  <div>
    <button
      className="btn-blue"
      onClick={[Function]}
    >
      Return
    </button>
  </div>
</div>
`;
